src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -89,18 +101,110 @@ function parseMessage(message, utf8Decoder) {
 
     return {
         ReferenceId,
+        MessageId,
         DataFormat,
         Data: data,
     };
 }
 
+function getNextRetryDelay(retryIndex) {
+    if (retryIndex < 0 || retryIndex > RECONNECT_DELAYS.length - 1) {
+        return null;
+    }
+
+    return RECONNECT_DELAYS[retryIndex];
+}
+
+function connect() {
+    let lastUsedToken = null;
+
+    this.connection = buildConnection({
+        baseUrl: this.baseUrl,
+        contextId: this.contextId,
+        messageId: this.lastMessageId,
+        accessTokenFactory: () => {
+            lastUsedToken = this.authToken;
+            return this.authToken;
+        },
+        protocol: this.protocol,
+    });
+
+    return this.connection.start().then(() => {
+        if (this.isDisconnecting) {
+            return;
+        }
+
+        this.createMessageStream(this.protocol);
+
+        this.connection.onclose((error) => {
+            // Do not try to re-connect if connection is explicitely closed or encountered error while starting streaming.
+            if (this.hasStreamingStarted && !this.isDisconnecting) {
+                reconnect.call(this, error);
+                return;
+            }
+
+            this.handleConnectionClosure(error);
+        });
+
+        // Token might have been updated while connect response was in flight
+        if (lastUsedToken !== this.authToken) {
+            this.renewSession();
+        }
+    });
+}
+
+function reconnect(error) {
+    this.isReconnecting = true;
+    this.stateChangedCallback(constants.CONNECTION_STATE_RECONNECTING);
+
+    log.debug(LOG_AREA, 'Attempting to reconnect', {
+        error,
+    });
+
+    // eslint-disable-next-line promise/no-promise-in-callback
+    retryConnection
+        .call(this, 0)
+        .then(() => {
+            this.isReconnecting = false;
+            this.stateChangedCallback(constants.CONNECTION_STATE_CONNECTED);
+        })
+        .catch((err) => {
+            log.debug(LOG_AREA, 'Failed to reconnect', {
+                error: err,
+            });
+
+            this.handleConnectionClosure();
+        });
+}
+
+function retryConnection(retryIndex, prevError) {

```

Comment:
I'm not in love with idea of introducing manual reconnect on top of what's implemented by signalr :( 
Few questions:

* Shouldn't oapi server be aware about last sent message and their 'received' status? Why they cannot calculate last message id themself? 
* Isn't there some 'factory' method for query building for reconnecting already built it into SignalrCore?

Reply:
server won't know if client is disconnected due to network issue so it won't have last received message.
I don't see any factory method for query build in new signalr core

Reply:
Although I see there is setter for baseUrl which I missed somehow and it can be changed during reconnection.
It should do the trick.
will update